test/node-rpc-test.js

```diff
@@ -225,4 +239,148 @@ describe('RPC', function() {
       }
     });
   });
+
+  describe('pruneblockchain', function() {
+    const network = Network.get(NETWORK);
+    const PRUNE_AFTER_HEIGHT = network.block.pruneAfterHeight;
+    const KEEP_BLOCKS = network.KEEP_BLOCKS;
+
+    let nclient, node;
+
+    before(() => {
+      network.block.pruneAfterHeight = 10;
+      network.block.keepBlocks = 10;
+    });
+
+    after(() => {
+      network.block.pruneAfterHeight = PRUNE_AFTER_HEIGHT;
+      network.block.keepBlocks = KEEP_BLOCKS;
+    });
+
+    afterEach(async () => {
+      if (nclient && nclient.opened)
+        await nclient.close();
+
+      if (node && node.opened)
+        await node.close();
+    });
+
+    it('should fail with wrong arguments', async () => {
+      node = new FullNode(nodeOptions);
+      nclient = new NodeClient(clientOptions);
+
+      await node.open();
+
+      await assert.rejects(async () => {
+        await nclient.execute('pruneblockchain', [1]);
+      }, {
+        code: errs.MISC_ERROR,
+        type: 'RPCError',
+        message: 'pruneblockchain'
+      });
+
+      await node.close();
+    });
+
+    it('should not work for spvnode', async () => {
+      node = new SPVNode(nodeOptions);
+      nclient = new NodeClient(clientOptions);
+
+      await node.open();
+
+      await assert.rejects(async () => {
+        await nclient.execute('pruneblockchain');
+      }, {
+        type: 'RPCError',
+        message: 'Cannot prune chain in SPV mode.',
+        code: errs.MISC_ERROR
+      });
+
+      await node.close();
+    });
+
+    it('should fail for pruned node', async () => {
+      node = new FullNode({
+        ...nodeOptions,
+        prune: true
+      });
+
+      await node.open();
+
+      await assert.rejects(async () => {
+        await nclient.execute('pruneblockchain');
+      }, {
+        type: 'RPCError',
+        code: errs.MISC_ERROR,
+        message: 'Chain is already pruned.'
+      });
+
+      await node.close();
+    });
+
+    it('should fail for short chain', async () => {
+      node = new FullNode(nodeOptions);
+
+      await node.open();
+
```

Comment:
Mine as well include assertion here that the chain is short -- a future dev may try adding blocks to the chain in an earlier test. Alternatively, you could add a `beforeEach()` or something that guarantees each `it` block starts with a fresh chain .

Reply:
There's is new FullNode in each test. These are `unitTest`s in a way.
`afterEach` close is there only if test fails and `await node.close` is not executed in the end of the tests.

Reply:
oh right, you got `new FullNode()` in each `it` nice.